### PR TITLE
docs: add Using Rspack build into build-performance

### PR DESCRIPTION
**Pull request by 9aoy**
## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/builder-doc/docs/zh/guide/optimization/build-performance.mdx [hunk 1]
```diff
@@ -36,6 +36,14 @@ nvm default 16
 node -v
 ```
 
+### 使用 Rspack 构建
+
+[Rspack](https://www.rspack.dev/) 是一个基于 Rust 的高性能 Web 构建工具，具备与 Webpack 生态系统的互操作性，可以被 Webpack 项目低成本集成，并提供更好的构建性能。
```
Comment (chenjiahan) [inline packages/document/builder-doc/docs/zh/guide/optimization/build-performance.mdx:41]: 这里直接 import shared/rspack.md 就好，避免维护多份 Rspack 描述

Comment (9aoy) [reply packages/document/builder-doc/docs/zh/guide/optimization/build-performance.mdx:41]: shared/rspack.md 那个 UI 是 tip，这个地方想用纯文本 😂

Comment (chenjiahan) [reply packages/document/builder-doc/docs/zh/guide/optimization/build-performance.mdx:41]: okk

Comment (chenjiahan) [reply packages/document/builder-doc/docs/zh/guide/optimization/build-performance.mdx:41]: 要不把 shared/rspack.md 改成 shared/rspack-tip.md

然后再加一份纯文本的 shared/rspack.md

Comment (9aoy) [reply packages/document/builder-doc/docs/zh/guide/optimization/build-performance.mdx:41]: ok

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**